openapiv3/annotations.proto

```diff
@@ -0,0 +1,52 @@
+// Copyright 2022 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package openapi.v3;
+
+import "openapiv3/OpenAPIv3.proto";
+import "google/protobuf/descriptor.proto";
+
+// This option lets the proto compiler generate Java code inside the package
+// name (see below) instead of inside an outer class. It creates a simpler
+// developer experience by reducing one-level of name nesting and be
+// consistent with most programming languages that don't support outer classes.
+option java_multiple_files = true;
+
+// The Java outer classname should be the filename in UpperCamelCase. This
+// class is only used to hold proto descriptor, so developers don't need to
+// work with it directly.
+option java_outer_classname = "AnnotationsProto";
+
+// The Java package name must be proto package name with proper prefix.
+option java_package = "org.openapi_v3";
+
+// A reasonable prefix for the Objective-C symbols generated from the package.
+// It should at a minimum be 3 characters long, all uppercase, and convention
+// is to use an abbreviation of the package name. Something short, but
+// hopefully unique enough to not conflict with things that may come along in
+// the future. 'GPB' is reserved for the protocol buffer implementation itself.
+option objc_class_prefix = "OAS";
+
+// The Go package name.
+option go_package = "./openapiv3;openapi_v3";
+
+extend google.protobuf.FileOptions {
+  Document document = 1042;
+}
+
+extend google.protobuf.MethodOptions {
+  Operation operation = 1042;
```

Comment:
Similarly, this merges an Operation into the operation generated for a method. This is a gnostic type (as is Document) - and I notice from your example that you're assigning them with valid structs, which are unintuitive to most OpenAPI users but fit the gnostic protobuf model - and I think that consistency is good. Without looking through your generator code (yet), I guess that makes it easier to merge these with the generated doc since that's working on gnostic protobuf-based structs.

Reply:
Protocol Buffer options need to be defined using protocol buffers themselves. So you will never get a 1-1 mapping of the options structure with OpenAPI structure as protocol buffers are strictly typed and yaml/json are not. Also, this is similar to how the [protoc-gen-openapiv2](https://github.com/grpc-ecosystem/grpc-gateway/tree/master/protoc-gen-openapiv2) generator works from the grpc-gateway project (though they use protocol buffers that are specific to OpenAPI v2/Swagger). I think most developers that will use this will understand this.

Since I need to use protocol buffer messages for the options, I reused the same ones that gnostic was already using to make merging them easier, as you said.